### ISSUE TITLE
Un-deprecate SpringBootStreamHandler:  FunctionInvoker will not work with an AWS Function Kafka Source

### ISSUE THREAD

**Issue by davidmelia** [spring-cloud/spring-cloud-function#943]
**Is your feature request related to a problem? Please describe.**
I have noticed in the latest spring cloud function (3.2.7) SpringBootStreamHandler is deprecated in favour of FunctionInvoker.

I have AWS Spring Cloud Function projects which are simple sources for emitting data to a Kafka Topic.  In these projects I simply extend SpringBootStreamHandler and define a 'function' bean as such 

```
import org.springframework.cloud.function.adapter.aws.SpringBootStreamHandler;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class MySpringBootStreamHandler extends SpringBootStreamHandler {

	public MySpringBootStreamHandler() {
		super(FunctionApplication.class);
		log.info("MySpringBootRequestHandler()");
	}

}

@Component(value = "function")
public class MyFunction implements Function<Flux<Map<String, String>>, Flux<String>> {
...
}
```
This means, programatically, I can specify the AWS function I want to run and the Spring Application main class.

Migrating to FunctionInvoker requires not only having to specify the main class outside of code (manifest) but also specifying a spring.cloud.function.definition.   spring.cloud.function.definition confuses spring cloud streams as it tries to bind to my AWS Lambda function.

**Describe the solution you'd like**
Either support SpringBootStreamHandler as a lower level route into spring cloud function AWS or allow FunctionInvoker to be overridden to provide the function name   i.e.

```
public class MyFunctionInvoker extends FunctionInvoker {

	public MyFunctionInvoker() {
		super("myFunctionName")
	}

}

```
N.B in fact it would be very useful to also specify the main class too i.e.

```
public class MyFunctionInvoker extends FunctionInvoker {

	public MyFunctionInvoker() {
		super("myFunctionName",FunctionApplication.class)
	}

}

```

**Describe alternatives you've considered**
Create and maintain my own Functioninvoker which is not ideal

**Additional context**
https://github.com/davidmelia/spring-cloud-function-source-only illustrates the issue.

- CallMySpringBootStreamHandler calls the deprecated SpringBootStreamHandler handler - all is well
- CallFunctionInvoker calls specifying spring.cloud.function.definition (spring streams gets confused) or not specifying spring.cloud.function.definition (spring function aws cannot establish a route)

**Comment by olegz** [comment]
@davidmelia So, `FunctionInvoker` is not final so you can extend it. But I believe what you are asking is that it could have an alternative for getting the function definition (other then `s.c.f.definition`), correct?

**Comment by davidmelia** [comment]
Exactly @olegz  - I'm sure there are more use cases than my example above so it would be super useful to be able to programmatically pass in the function definition (and the main class if poss).  This would be a great fallback.

**Comment by olegz** [comment]
I buy that. 
Glad you mentioned it now before the releases. Will address it tomorrow

**Comment by olegz** [comment]
@davidmelia It's done, there is a new constructor that would take function definition as a parameter and that would take precedence over property-based definition. It was backported to 3.2.x, snapshots are available (3.2.8-SNAPSHOT). Please give it a shot and let me know as we will be releasing 3.2.8 tomorrow.

**Comment by davidmelia** [comment]
@olegz just tested and works perfectly.  Thanks
